packages/debug/src/browser/debug-configuration-manager.ts

```diff
@@ -258,18 +254,13 @@ export class DebugConfigurationManager {
         });
     }
     protected async doCreate(model: DebugConfigurationModel): Promise<URI> {
-        const uri = new URI(model.workspaceFolderUri).resolve('.theia/launch.json');
         const debugType = await this.selectDebugType();
         const configurations = debugType ? await this.provideDebugConfigurations(debugType, model.workspaceFolderUri) : [];
         const content = this.getInitialConfigurationContent(configurations);
-        try {
-            await this.filesystem.createFile(uri.toString(), { content });
-        } catch (e) {
-            if (!FileSystemError.FileExists.is(e)) {
-                throw e;
-            }
-        }
-        return uri;
+        await this.preferences.set('launch', content);
+        const { configUri } = this.preferences.resolve('launch');
+        // @ts-ignore
```

Comment:
Is the `// @ts-ignore` necessary?
The variable `configUri` may be undefined which does not respect the return type of the method.
Should we handle this more gracefully than suppressing the warning?

Reply:
now there is a fallback for this case